fix: BROS-14: Optimize yarn test:e2e

niklub opened the pull request:
---

## Summary of the Pull Request

This Pull Request introduces the following changes across multiple components related to testing and error handling:
Test Workflow Update: Modified GitHub Actions workflow to run end-to-end tests using a more specific command (yarn lsf:e2e:ci).

### Workflow Update:

- Modified .github/workflows/tests-yarn-e2e.yml to use CI specified running command.

### AudioUltra fixes

- Error rendering logic moved to the parent component to be able to be relevant thanks to the observer.

### Testing Improvements:

- Remove unnecessary waitForAction
- Replaced `I.wait` with `I.waitTicks` for more readability and understandability
- Added execution waiting to tests where it's needed
- Reuse helpers more often
- Temporary skip outdated tests
- Introduced timeout mechanisms for asynchronous audio loading.
- Move some tests to Cypress to avoid running them in parallel which makes them flaky

### Helper and Utility Enhancements:
- Added `seeFocusedElement` and `hasError` methods to improve usability for test writing.
- Updated Playwright and other test helpers with additional configuration.
- Fix `waitForAudio` conditions

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Commit:
fix: BROS-14: Optimize yarn test:e2e

niklub commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15024410666)
> Successfully merged:  create mode 100644 .github/workflows/apply-linters.yml

niklub committed:
Merge branch 'develop' into 'fb-bros-14'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15024410666

Commit:
Add parallel execution in CI

Commit:
Cleanup

Commit:
try: Await for LS destroy

Commit:
try fixing smoke tests

Gondragos committed:
Merge branch 'develop' into fb-bros-14

Gondragos committed:
fix: fix e2e tests

Gondragos committed:
test: skip fully broken e2e test scenarios

Gondragos committed:
linting fixes

Gondragos committed:
fix: configure waitForAction in audio errors test

Gondragos committed:
test: skip flacky failing audio error scenarios

Temporarily skipped two audio error handling tests to prevent false negatives.

Gondragos committed:
fix: update lsf:e2e:parallel script command

Gondragos committed:
Upgrade Playwright dependency to v1.52.0

Gondragos committed:
fix: update e2e workflow to use correct command

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15181918674)
> Successfully merged:  create mode 100644 web/libs/ui/src/lib/Tooltip/tooltip.stories.tsx

Gondragos committed:
Merge branch 'develop' into 'fb-bros-14'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15181918674

Gondragos committed:
Revert "Upgrade Playwright dependency to v1.52.0"

This reverts commit 65b2a484adbbd77cf64c26e21dc43dd2c47c6724.

Gondragos committed:
Merge commit '1a4fb8bab02b5884bcc101bbcfa63923ba9bf875' into fb-bros-14

Gondragos committed:
fix: rework focused element detection logic

hlomzik commented on web/libs/editor/tests/e2e/codecept.conf.js:
```diff
@@ -21,14 +21,16 @@ module.exports.config = {
-      timeout: 60000, // Action timeout after 60 seconds
-      waitForAction: headless ? 300 : 1200,
+      timeout: 30000,
+      waitForAction: headless ? 0 : 0,
```
there should be some delay in non-headless more so humans can track events

Gondragos replied:
```suggestion
      waitForAction: headless ? 0 : 500,
```

Commit:
Revert testing values

Gondragos committed:
refactor: migrate audio error tests to Cypress

Replaced legacy Playwright-based audio error tests with Cypress.
- Introduced `hasError` helper in `AudioView` for streamlined error message assertions.

Gondragos committed:
Merge remote-tracking branch 'origin/fb-bros-14' into fb-bros-14

Gondragos committed:
refactor: replace explicit waits with `waitTicks`

Replaced some `I.wait` calls in e2e tests with the new `I.waitTicks` method for consistency and improved readability. Introduced the `waitTicks` utility in `steps_file.js` as a scalable alternative to explicit time-based waits.

Changes include:
- Updated wait calls across multiple test files and fragments to use `waitTicks`.
- Fixed a typo in a comment ("chanve" to "chance") in `image-list.test.js`.
- Adjusted wait durations where necessary to align with the `waitTicks` scale.

Gondragos committed:
test: fix formatting in audio error test

Gondragos committed:
test: revert wait time for magic wand test

Gondragos committed:
fix: adjust wait timings

Gondragos committed:
test: add behavior based wait

Gondragos committed:
fix: add custom timeout to allow to process loading audio

Gondragos committed:
fix: correct region count validation in test

Gondragos committed:
fix: AudioUltra errors rendering

Gondragos committed:
fix: fix modal handling in e2e tests

Gondragos committed:
test: Fix DateTime validation test with Modals helper

Gondragos committed:
fix: add wait to stabilize text selection

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15254104277)
> Successfully merged:  create mode 100644 web/apps/playground/src/hooks/useCopyText.ts

Gondragos committed:
Merge branch 'develop' into 'fb-bros-14'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15254104277

Gondragos committed:
test: configure waitForAction in shortcuts test

Gondragos committed:
Merge remote-tracking branch 'origin/fb-bros-14' into fb-bros-14

Gondragos committed:
fix: ensure stability by adding waitTicks in audio tests

Gondragos committed:
fix: ensure modal stability with additional wait

hlomzik commented on web/libs/editor/tests/e2e/steps_file.js:
```diff
@@ -22,4 +22,8 @@ module.exports = () =>
       this._performActionEnd(name);
     },
+
+    waitTicks(n) {
```
Love this approach and name!

hlomzik commented on web/libs/editor/tests/e2e/tests/image.zoom-rotate.test.js:
What are we waiting for here? Could it also be `waitTicks()`?

Gondragos replied:
In this case ticks were not a stable solution. Somehow it needs more time to wait.

hlomzik reviewed (changes requested):
Add `todo` comments to all skipped tests, so we'll know what to do with them later. The rest is great, just couple of small questions!

Gondragos committed:
test: mark outdated e2e tests with TODOs and skip

Annotated and skipped three e2e tests due to outdated or problematic functionality:
- Taxonomy read-only in annotation history.
- Sync issues with multiple audio playback.
- Removing textarea items in 'region-list' mode.
Each test now includes a TODO comment to clarify necessary actions for future fixes.

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15275674435)
> Successfully merged:  create mode 100644 docs/themes/v2/source/images/import-export/keypoints.png

Gondragos committed:
Merge branch 'develop' into 'fb-bros-14'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15275674435
